test/e2e/set/scyllacluster/scyllamanager.go

```diff
@@ -267,82 +270,470 @@ var _ = g.Describe("Scylla Manager integration", func() {
 		o.Expect(tasks.TaskListItemSlice).To(o.BeEmpty())
 	})
 
-	g.It("should register cluster and sync backup tasks", func() {
+	provideGCSCredentials := func(ctx context.Context, sc *scyllav1.ScyllaCluster) *scyllav1.ScyllaCluster {
+		scCopy := sc.DeepCopy()
+
+		gcServiceAccountKey := f.GetGCSServiceAccountKey()
+		o.Expect(gcServiceAccountKey).NotTo(o.BeEmpty())
+
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "gcs-service-account-key-",
+			},
+			Data: map[string][]byte{
+				"gcs-service-account.json": gcServiceAccountKey,
+			},
+		}
+
+		secret, err := f.KubeClient().CoreV1().Secrets(f.Namespace()).Create(ctx, secret, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		for i := range scCopy.Spec.Datacenter.Racks {
+			scCopy.Spec.Datacenter.Racks[i].Volumes = append(scCopy.Spec.Datacenter.Racks[i].Volumes, corev1.Volume{
+				Name: "gcs-service-account",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: secret.Name,
+					},
+				},
+			})
+			scCopy.Spec.Datacenter.Racks[i].AgentVolumeMounts = append(scCopy.Spec.Datacenter.Racks[i].AgentVolumeMounts, corev1.VolumeMount{
+				Name:      "gcs-service-account",
+				ReadOnly:  true,
+				MountPath: "/etc/scylla-manager-agent",
+			})
+		}
+
+		return scCopy
+	}
+
+	g.It("should register cluster, sync backup tasks and support manual restore procedure", func() {
 		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
 		defer cancel()
 
-		sc := f.GetDefaultScyllaCluster()
-		sc.Spec.Datacenter.Racks[0].Members = 1
+		sourceSC := f.GetDefaultScyllaCluster()
+		sourceSC.Spec.Datacenter.Racks[0].Members = 1
 
-		framework.By("Creating a ScyllaCluster")
-		sc, err := f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
+		objectStorageType := f.GetObjectStorageType()
+		switch objectStorageType {
+		case framework.ObjectStorageTypeGCS:
+			sourceSC = provideGCSCredentials(ctx, sourceSC)
+		default:
+			g.Fail("unsupported object storage type")
```

Comment:
it should skip the test if it's not provided at all

Reply:
It will be skipped silently on misconfiguration then, not desired imo.